### PR TITLE
Setting up a "Hello world" project as one app with Webpack

### PR DESCRIPTION
**Pull request by kiboma2021**
**Project requirements**

1. Create a new Rails app called 'hello-rails-react'.
2. Use the command rails new -j webpack my_new_rails_app
3. Verify that the jsbundling-rails gem is added to your Gemfile.
4. Initialize your project with Git.
5. Install needed node dependencies.
6. Create a babel.config.js in the root folder of your project.
7. Update webpack.config.js to include the babel loader.
8. Make sure that your project has a Postgres database set up.
9. Create a table for storing your messages and create 5 different greetings.
10. Create an API endpoint that selects a random greeting from your table (you will need a controller with an action and Rails route).
11. Create a static view that will be the root of your app.
12. Create the App component with react-router and render it in your static view.
13. Create the Greeting component that will display a greeting. Set it up as a route in your App component.
14. Create a store, an action, and a reducer that will connect to your API endpoint to get the random greeting.
15. Display the random greeting in your Greeting component.

### CODE REVIEW

**Comment by KZ-ranjeet** [comment]
@kiboma2021 check slack for my msg

**Review by omar-muhamad — COMMENTED** [review]
# Invalid ❌

Hi @kiboma2021, 

#### While you made a great effort in this project, unfortunately, I cannot proceed to review your code.

#### Your Code Review Request is marked as invalid for this reason:

- [ ] Linters GitHub actions are not working. Please configure it.
- [ ] The MIT link in the README file is broken, Please add a valid link.


Your Code Review Request will be marked as invalid in your Dashboard, so please submit a new one once you are ready 🙏

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

--------

_Invalid Code Review Request **Does not count** into the code reviews limit._

File: app/javascript/react/src/components/Home.jsx [hunk 1]
```diff
@@ -0,0 +1,26 @@
...
+  const [data, getData] = React.useState([]);
+  const url = 'http://127.0.0.1:3000/random_greeting';
+  React.useEffect(() => {
+    async function fetchGreeting() {
+      const response = await fetch(url);
+      const result = await response.json();
+      getData(result);
+    }
+    fetchGreeting();
+  }, [url]);
```
Comment (omar-muhamad) [inline app/javascript/react/src/components/Home.jsx]: - [ ] As per this [requirement](https://github.com/microverseinc/curriculum-rails/blob/main/connect-frontend-frameworks/rails_react_webpack.md#:~:text=Create%20a%20store%2C%20an%20action%2C%20and%20a%20reducer%20that%20will%20connect%20to%20your%20API%20endpoint%20to%20get%20the%20random%20greeting.), please use redux toolkit and create a store, an action, and a reducer that will connect to your API endpoint to get the random greeting, not directly fetch the data inside `useEffect` as you did.

File: README.md [hunk 2]
```diff
@@ -1,24 +1,243 @@
...
+<p align="right">(<a href="#readme-top">back to top</a>)</p>
+
+---
+
+<!-- LICENSE -->
+
+## 📝 License <a name="license"></a>v
+
+This project is [MIT](https://web.mit.edu/) licensed.
+
```
Comment (omar-muhamad) [inline README.md]: - [ ] Please add the MIT licence link, not the MIT site link as you did.

**Review by omar-muhamad — CHANGES_REQUESTED** [review]
# Changes Required ♻️
### Hi @kiboma2021,

### Nice job, you are almost there.

cat-driving-serious

## (Highlights) Good Points: :+1:
- Your work is well documented. ✔
- The correct workflow is used.  ✔
- All linters checks are ok. ✔

## (Changes Required) Aspects to improve: ♻️

- [ ] Please check the inline comments below ⬇️.

## [Optional] suggestions:

- Nothing to mention.

_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

**Cheers and Happy coding!👏👏👏**

Feel free to leave any questions or comments in the PR thread if something is not 100% clear, and please remember to tag me in your question so I can receive the notification**

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s).

**Review by Reem-lab — APPROVED** [review]
# Hi @kiboma2021,  👋🏻👋🏻

Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉 


## Highlights

- [ ]  Descriptive PR⭐
- [ ]  No Linters error 🤩
- [ ] Good job working on the API💪🏻
- [ ] Amazing for working on the requested changes⭐



## Optional suggestions

_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will really improve your application._

Cheers, and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**